app/models/iam_user_identity.rb

```diff
@@ -52,4 +59,14 @@ def set_expire
   def uuid
     Digest::UUID.uuid_v5(@iam_sec_id, @iam_icn)
   end
+
+  # Return a single mhv id from a possible comma-separated list value attribute
+  def self.valid_mhv_id(id_from_profile)
+    # TODO: site login fails if multiple _different_ MHV IDs are present
+    # Should this code be doing the same? Or at least logging a warning?
+    # see lib/saml/user_attributes/ssoe.rb for example validation
```

Comment:
I feel like we should resolve these comments before merging.

Reply:
It's not clear  what the right thing to do is. On VA.gov, when this  occurs we prevent sign-in, and we have the opportunity to present a clear  error  message during the sign-in flow. 

Here, IAM is going to issue the access token successfully. We can prevent this API request, but we could be instantiating this user identity object for any of the OAuth-based API endpoints - it's not like there's a first API  request  that always happens after token issuance. That means any part of the app that makes an API request needs to be prepared to handle getting an error back and presenting something meaningful to the user. And we need to define what that error is.

Would you be okay if we filed a launch-blocker ticket and revisited it, in the interest of  getting the MHV integration moving  along?

Reply:
That sounds good, thanks Patrick. For the sake of this PR, let's log a warning and at least we have some visibility into how often this occurs.

Reply:
Do we want to just log a warning or create a Crashlytics event for this?

Reply:
I guess probably not crashlytics here, but is there a way to make it easy to pull a report on this?

Reply:
Two other mitigation options:

1) Get the IAM team to alter their logic such that they don't issue an access token if they detect multiple MHV IDs. But what if the  second  MHV ID is added _after_ the refresh token is issued? Complex.

2) Just gate access to MHV features such as SM - i.e. remove the MHV ID from the session, and  thus secure messaging will disappear from the authorizedServices list. There's no reason to gate access to claims becaause of a duplicate  MHV ID, for example.  Might need a "why can't I  access?" UX  consideration though.

Reply:
I filed this issue for follow-up: https://github.com/department-of-veterans-affairs/va.gov-team/issues/19971

Reply:
We can also log a sentry event for it for visibility.